### PR TITLE
added fqdn type, which handles internationalized domain names

### PR DESCRIPTION
**Pull request by vertexmc**

### CODE REVIEW

File: synapse/models/inet.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
+from encodings import idna
+import hashlib
```
Comment (invisig0th) [inline synapse/models/inet.py]: lulz.  visi OCD...  move me back down plz!

File: synapse/models/inet.py [hunk 2]
```diff
@@ -192,15 +194,19 @@ def onTufoFormFqdn(mesg):
...
             if pafo[1].get('inet:fqdn:sfx'):
                 props['inet:fqdn:zone'] = 1
 
-    # FIXME this needs to mark/unmark kids as zone
-    #def onTufoSetFqdnSfx(mesg):
+    def onTufoSetFqdnSfx(mesg):
+        sfx = mesg[1].get('valu')
+        fqdn = mesg[1].get('tufo')[1].get('inet:fqdn')
+        for tufo in core.getTufosByProp('inet:fqdn:domain', fqdn):
+            core.setTufoProp(tufo, 'zone', sfx)
```
Comment (invisig0th) [inline synapse/models/inet.py]: nice.  <3 lack of conditional logic here via reuse of sfx.

File: synapse/models/inet.py (hunk 1 above)
Comment (vertexmc) [inline synapse/models/inet.py]: moved

File: synapse/lib/types.py [hunk 3]
```diff
@@ -88,7 +88,7 @@ def parse(self, text, oldval=None):
         return self.norm(text, oldval=oldval)
 
     def repr(self, valu):
-        return str(valu)
+        return valu
```
Comment (vertexmc) [inline synapse/lib/types.py]: added unit test demonstrating this doesn't break repr for ints

File: synapse/tests/test_model_inet.py [hunk 4]
```diff
@@ -72,3 +66,107 @@ def test_model_inet_asnet4(self):
...
+            valu = '%s@%s' % ('user', 'xn--tst.xampl.link-vjbdf')
+            tufo = core.formTufoByProp(prop, valu)
+            self.eq(tufo[1].get('inet:email:fqdn'), 'tèst.èxamplè.link')
+            self.eq(tufo[1].get('inet:email'), 'user@tèst.èxamplè.link')
+
+        with s_cortex.openurl('ram:///') as core:
+            prop = 'inet:url'
+            valu = '%s://%s/%s' % ('https', 'xn--tst.xampl.link-vjbdf', 'things')
+            tufo = core.formTufoByProp(prop, valu)
+            self.eq(tufo[1].get('inet:url'), 'https://xn--tst.xampl.link-vjbdf/things') # hostpart is not normed in inet:url
```
Comment (vertexmc) [inline synapse/tests/test_model_inet.py]: @invisig0th - the host part of the url is not normed

**Pull request merged**